apps/hash-ai-worker-ts/src/activities/flow-activities/research-entities-action/open-ai-coordinating-agent.ts

```diff
@@ -257,28 +259,41 @@ const parseCoordinatorInputs = async (params: {
 
   const { userAuthentication } = await getFlowContext();
 
+  /**
+   * @todo: simplify the properties in the existing entities
+   */
+  const existingEntities = inputExistingEntities
+    ? mapActionInputEntitiesToEntities({ inputEntities: inputExistingEntities })
+    : undefined;
+
   const dereferencedEntityTypes = await getDereferencedEntityTypesActivity({
     graphApiClient,
-    entityTypeIds: entityTypeIds!,
+    entityTypeIds: [
+      ...entityTypeIds!,
+      /**
+       * We need to include the types of the existing entities, to correctly
+       * determine whether the links are satisfiable.
+       *
+       * @todo: address this in the `getDereferencedEntityTypesActivity` function
+       */
```

Comment:
Are you planning to address this in the near-ish future? If not I'd move the `@todo` into the function itself and create a task for it (you may already have done so)

Reply:
Yeah I think we should address it, had a quick stab at this but ended up being more complex than expected. I've created [a ticket](https://linear.app/hash/issue/H-2685/in-getdereferencedentitytypes-allow-for-specifying-additional-entity), moved the comments into the method and linked the ticket in https://github.com/hashintel/hash/pull/4399/commits/47d45bc16efe53d3f5cd8f72627cf35f3fc35c72.